### PR TITLE
 Google Play extension refactoring

### PR DESCRIPTION
**Pull request by xzhe**
Refactors have done:
1. Add type information.
2. Define interface to replace object initializer.
3. Remove all global variables, passing objects as arguments between sub functions.
4. Use gulp build framework.
5. Localize string.
6. tslint.
7. Add more checks in the authentication part. Throw an error if there is any problem.

### CODE REVIEW

File: Tasks/google-play-promote/task.loc.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    "Build",
+    "Release"
+  ],
+  "demands": [
+    "npm"
+  ],
+  "version": {
+    "Major": "1",
+    "Minor": "3",
+    "Patch": "16"
```
Comment (DavidStaheli) [inline Tasks/google-play-promote/task.loc.json]: This version should be 1.112.0 too.

File: Tasks/google-play-release/task.loc.json [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    "Build",
+    "Release"
+  ],
+  "demands": [
+    "npm"
+  ],
+  "version": {
+    "Major": "1",
+    "Minor": "3",
+    "Patch": "16"
```
Comment (DavidStaheli) [inline Tasks/google-play-release/task.loc.json]: This version should be 1.112.0 too.

File: Tasks/google-play-promote/google-play-promote.ps1 [hunk 3]
```diff
@@ -14,4 +14,4 @@ $env:INPUT_sourceTrack = $sourceTrack
 $env:INPUT_destinationTrack = $destinationTrack
 $env:INPUT_userFraction = $userFraction
 
-node google-play-promote.js
+node ../../_build/Tasks/GooglePlayPromote/google-play-promote.js
```
Comment (DavidStaheli) [inline Tasks/google-play-promote/google-play-promote.ps1]: The _build folder doesn't exist in the extension package (.vsix file), does it?

File: Tasks/google-play-release/GooglePlay.ps1 [hunk 4]
```diff
@@ -14,4 +14,4 @@ $env:INPUT_track = $track
 $env:INPUT_userFraction = $userFraction
 $env:INPUT_changeLogFile = $changeLogFile
 
-node GooglePlay.js
+node ../../_build/Tasks/GooglePlayRelease/GooglePlay.js
```
Comment (DavidStaheli) [inline Tasks/google-play-release/GooglePlay.ps1]: The _build folder doesn't exist in the extension package (.vsix file), does it?

File: Tasks/google-play-rollout-update/google-play-rollout-update.ps1 [hunk 5]
```diff
@@ -10,4 +10,4 @@ $env:INPUT_serviceEndpoint = $serviceEndpoint
 $env:INPUT_packageName = $packageName
 $env:INPUT_userFraction = $userFraction
 
-node google-play-rollout-update.js
+node ../../_build/Tasks/GooglePlayIncreaseRollout/google-play-rollout-update.js
```
Comment (DavidStaheli) [inline Tasks/google-play-rollout-update/google-play-rollout-update.ps1]: The _build folder doesn't exist in the extension package (.vsix file), does it?

File: Tasks/google-play-rollout-update/google-play-rollout-update.ts [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+import tl = require('vsts-task-lib/task');
+let google = require('googleapis');
+let publisher = google.androidpublisher('v2');
+
+interface ClientKey {
+    client_email?: string;
+    private_key?: string;
+}
+
+interface AndroidResourse {
```
Comment (DavidStaheli) [inline Tasks/google-play-rollout-update/google-play-rollout-update.ts]: This should be AndroidResource instead of AndroidResourse.

File: Tasks/google-play-rollout-update/task.loc.json [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    "Build",
+    "Release"
+  ],
+  "demands": [
+    "npm"
+  ],
+  "version": {
+    "Major": "1",
+    "Minor": "3",
+    "Patch": "16"
```
Comment (DavidStaheli) [inline Tasks/google-play-rollout-update/task.loc.json]: This version should be 1.112.0 too.

File: package.json [hunk 8]
```diff
@@ -2,9 +2,34 @@
   "name": "vso-android-extension",
   "version": "1.3.0",
```
Comment (DavidStaheli) [inline package.json]: This version should be 1.112.0 to match the task versions.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
+# Visual Studio Team Services Extension for the Apple App Store Contributor Guide
+The instructions below will help you set up your development environment to contribute to this repository.
+Make sure you've already cloned the repo.  :smile:
+
+## Ways to Contribute
+Interested in contributing to the vsts-vscode project? There are plenty of ways to contribute, all of which help make the project better.
```
Comment (xzhe) [inline CONTRIBUTING.md]: Done!

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+* Submit a [bug report](https://github.com/Microsoft/app-store-vsts-extension/issues/new) or [feature request](https://github.com/Microsoft/app-store-vsts-extension/issues/new) through the Issue Tracker
+* Review the [source code changes](https://github.com/Microsoft/app-store-vsts-extension/pulls)
+* Submit a code fix for a bug (see `Submitting Pull Requests` below)
+* Participate in [discussions](https://github.com/Microsoft/app-store-vsts-extension/issues)
```
Comment (xzhe) [inline CONTRIBUTING.md]: Done!

File: Tasks/google-play-promote/google-play-promote.ts [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+import Promise = require('bluebird');
+let google = require('googleapis');
+let publisher = google.androidpublisher('v2');
+
+interface ClientKey {
+    client_email?: string;
+    private_key?: string;
+}
+
+interface AndroidResourse {
```
Comment (xzhe) [inline Tasks/google-play-promote/google-play-promote.ts]: Done!

File: Tasks/google-play-promote/task.loc.json (hunk 1 above)
Comment (xzhe) [inline Tasks/google-play-promote/task.loc.json]: Done!

File: Tasks/google-play-release/task.loc.json (hunk 2 above)
Comment (xzhe) [inline Tasks/google-play-release/task.loc.json]: Done!

File: Tasks/google-play-rollout-update/google-play-rollout-update.ts (hunk 6 above)
Comment (xzhe) [inline Tasks/google-play-rollout-update/google-play-rollout-update.ts]: Done!

File: Tasks/google-play-rollout-update/task.loc.json (hunk 7 above)
Comment (xzhe) [inline Tasks/google-play-rollout-update/task.loc.json]: Done!

File: package.json (hunk 8 above)
Comment (xzhe) [inline package.json]: Done!

File: Tasks/google-play-promote/google-play-promote.ps1 (hunk 3 above)
Comment (xzhe) [inline Tasks/google-play-promote/google-play-promote.ps1]: I revert the change, thanks!

File: Tasks/google-play-release/GooglePlay.ps1 (hunk 4 above)
Comment (xzhe) [inline Tasks/google-play-release/GooglePlay.ps1]: I revert the change, thanks!

File: Tasks/google-play-rollout-update/google-play-rollout-update.ps1 (hunk 5 above)
Comment (xzhe) [inline Tasks/google-play-rollout-update/google-play-rollout-update.ps1]: I revert the change, thanks!
